atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/workflow/TableConfiguration.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.workload.workflow;
+
+import com.palantir.atlasdb.transaction.api.ConflictHandler;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface TableConfiguration {
+    String tableName();
+
+    ConflictHandler conflictHandler();
```

Comment:
I don't think that this is json serializable/deserializable -- can we make it so? Just b/c I anticipate we need to move towards a world of config-driven workflows sooner rather than later, with the desire to use this internally as a standalone service.

Reply:
Enums are JSON (de)serializable as their name, so this should be fine? Unless you mean the entire class - which is fair, will add.

Reply:
Yeah I was being silly, you can ignore this lol.

Reply:
I introduce in #6471 an `IsolationLevel` interface, which we can use here instead, and then move this class to the api layer.